### PR TITLE
[Migration prep] Docs: "QL language reference" category

### PR DESCRIPTION
**Pull request by shati-patel**
## Summary

Closes https://github.com/github/semmle-docs/issues/21.

### Changes in this PR

| Original | Updated | Source | Notes |
| --- | --- | --- | --- |
| [QL handbook landing page](https://help.semmle.com/QL/ql-handbook)  | [QL language reference category overview](http://docteam.internal.semmle.com/shati/ql-handbook/) | [Source](https://github.com/Semmle/ql/pull/3178/files#diff-b0785d87c60423fc1a1db439ecf8f620) | Add intro, remove other text and links |
| [About QL](https://help.semmle.com/QL/learn-ql/about-ql.html) | [About the QL language](http://docteam.internal.semmle.com/shati/ql-handbook/about-the-ql-language.html) | [Source](https://github.com/Semmle/ql/pull/3178/files#diff-24eb046f6e634a5fbb6f695a7273d588) | Move into QL language category and add intro |
| [QL specifications](https://help.semmle.com/QL/ql-spec/) (landing page and child topics) | _deleted_ | [Source](https://github.com/Semmle/ql/pull/3178/files#diff-b72ce8f355f6fdebe652f016daf638f0) | I haven't fully deleted the Sphinx project yet as I suspect that'll break the PR checks. Added a README note for now. |

The other changes are formatting fixes and link updates. 

### Outstanding work

This restructuring will have some impact when we republish the documentation. I'll raise an issue with details, but I'm keeping note here for now:

- Replace "QL language handbook" with "QL language reference" throughout docs.
- Remove the QL spec from the Jenkins `docs/sphinx` build target.
- Redirect "About QL"
- Redirect the QL specifications and their landing page
- Update https://help.semmle.com/QL/ql-reference-topics.html

### CODE REVIEW

File: docs/language/ql-handbook/about-the-ql-language.rst [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
+######################
 
-This section is aimed at users with a background in general purpose programming as well as in databases. For a basic introduction and information on how to get started, see :doc:`Introduction to QL <introduction-to-ql>` and :doc:`Learning CodeQL <../index>`.
+QL is the powerful query language that underlies CodeQL, which is used to analyze code.
 
-QL is a declarative, object-oriented query language that is optimized to enable efficient analysis of hierarchical data structures, in particular, databases representing software artifacts.
-
-The queries and metrics used in LGTM are implemented using CodeQL, which uses QL to analyze code. This ensures that they can be extended or revised easily to keep up with changes in definitions of best coding practice. We continually improve existing queries as we work towards the ultimate goal of 100% precision.
+For more information about the important concepts and syntactic constructs of QL, see the individual reference topics such as :doc:`Predicates <predicates>` and :doc:`Types <types>`.
+The explanations and examples help you understand how the language works, and how to write more advanced QL code.
```
Comment (felicitymay) [inline docs/language/ql-handbook/about-the-ql-language.rst:7]: I'm not sure whether this is the right place for these links. Possibly we could leave the TOC in the sidebar to advertise the first set of links? Perhaps we could link to the specifications from the "Properties of QL" section?

File: docs/language/ql-handbook/about-the-ql-language.rst [hunk 2]
```diff
@@ -56,14 +59,12 @@ Here are a few prominent conceptual and functional differences between general p
...
 -  In object oriented programming languages, instantiating a class involves creating an object by allocating physical memory to hold the state of that instance of the class. In QL, classes are just logical properties describing sets of already existing values.
 
-These topics are discussed in detail in the `QL language handbook <https://help.semmle.com/QL/ql-handbook/index.html>`__.
-
-References
-----------
+Further reading
+---------------
 
 Academic references available from the `Semmle website <https://help.semmle.com/publications.html>`__ also provide an overview of QL and its semantics. Other useful references on database query languages and Datalog:
```
Comment (felicitymay) [inline docs/language/ql-handbook/about-the-ql-language.rst:65]: Probably should have updated this link name while we were updating the link. It doesn't really make sense now that it's all on one site.
```suggestion
`Academic references <https://help.semmle.com/publications.html>`__ also provide an overview of QL and its semantics. Other useful references on database query languages and Datalog:
```

File: docs/language/ql-handbook/about-the-ql-language.rst (hunk 1 above)
Comment (shati-patel) [reply docs/language/ql-handbook/about-the-ql-language.rst]: That's a good point. I've moved the links down to the "properties" section and changed them so they're relevant to the code example.

**Comment by shati-patel** [comment]
Thanks for the review and the suggestions, @felicitymay 😄

**Comment by shati-patel** [comment]
Woops, I forgot to update the name that's displayed in the sidebar TOC and the page title!
Fixed in that last commit.

**Comment by felicitymay** [comment]
It's a good thing you're on the ball today, I completely missed those issues 🙍 
:shipit:  when you're ready 😄

**Pull request merged**